common/src/View/CreateBrushToolBase.cpp

```diff
@@ -65,45 +71,75 @@ void CreateBrushToolBase::createBrush()
 
 void CreateBrushToolBase::cancel()
 {
-  m_brushNode.release();
+  for (auto& node : m_brushNodes)
+  {
+    node.release();
+  }
 }
 
 void CreateBrushToolBase::render(
   Renderer::RenderContext& renderContext, Renderer::RenderBatch& renderBatch)
 {
-  if (m_brushNode)
+  if (m_brushNodes.size() > 0)
   {
-    renderBrush(renderContext, renderBatch);
+    renderBrushes(renderContext, renderBatch);
   }
 }
 
-void CreateBrushToolBase::renderBrush(
+void CreateBrushToolBase::renderBrushes(
   Renderer::RenderContext& renderContext, Renderer::RenderBatch& renderBatch)
 {
-  ensure(m_brushNode, "brush is not null");
-
-  m_brushRenderer->setFaceColor(pref(Preferences::FaceColor));
-  m_brushRenderer->setEdgeColor(pref(Preferences::SelectedEdgeColor));
-  m_brushRenderer->setShowEdges(true);
-  m_brushRenderer->setShowOccludedEdges(true);
-  m_brushRenderer->setOccludedEdgeColor(Color(
-    pref(Preferences::SelectedEdgeColor), pref(Preferences::OccludedSelectedEdgeAlpha)));
-  m_brushRenderer->setTint(true);
-  m_brushRenderer->setTintColor(pref(Preferences::SelectedFaceColor));
-  m_brushRenderer->setForceTransparent(true);
-  m_brushRenderer->setTransparencyAlpha(0.7f);
-
-  m_brushRenderer->clear();
-  m_brushRenderer->addBrush(m_brushNode.get());
-  m_brushRenderer->render(renderContext, renderBatch);
-
-  auto boundsRenderer = Renderer::SelectionBoundsRenderer{m_brushNode->logicalBounds()};
-  boundsRenderer.render(renderContext, renderBatch);
+  for (size_t i = 0; i < m_brushNodes.size(); ++i)
+  {
+    auto brushNode = m_brushNodes[i].get();
+    ensure(brushNode, "brush is not null");
+
+    m_brushRenderer->setFaceColor(pref(Preferences::FaceColor));
+    m_brushRenderer->setEdgeColor(pref(Preferences::SelectedEdgeColor));
+    m_brushRenderer->setShowEdges(true);
+    m_brushRenderer->setShowOccludedEdges(true);
+    m_brushRenderer->setOccludedEdgeColor(Color(
+      pref(Preferences::SelectedEdgeColor), pref(Preferences::OccludedSelectedEdgeAlpha)));
+    m_brushRenderer->setTint(true);
+    m_brushRenderer->setTintColor(pref(Preferences::SelectedFaceColor));
+    m_brushRenderer->setForceTransparent(true);
+    m_brushRenderer->setTransparencyAlpha(0.7f);
+
+    m_brushRenderer->clear();
+    m_brushRenderer->addBrush(brushNode);
```

Comment:
Do everything outside of the loop except adding the brushes. I think there might even be `BrushRenderer::addBrushes`.